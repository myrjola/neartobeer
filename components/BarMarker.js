import React, { PropTypes } from 'react';
import MapView from 'react-native-maps';

import { BEER_UNDER_30, BEER_UNDER_40, BEER_UNDER_50 } from '../reducers/maxBeerPriceCategory';

import greenmug from '../images/GreenMug.png';
import yellowmug from '../images/YellowMug.png';
import redmug from '../images/RedMug.png';

export function chooseBeerIcon(maxBeerPriceCategory) {
  switch (maxBeerPriceCategory) {
    case BEER_UNDER_30:
      return greenmug;
    case BEER_UNDER_40:
      return yellowmug;
    default:
    case BEER_UNDER_50:
      return redmug;
  }
}

export const getBarCoordinate = bar => ({
  latitude: bar.post_latitude,
  longitude: bar.post_longitude,
});

const BarMarker = ({ bar, onPress }) => (
  <MapView.Marker
    title={bar.post_title}
    description={bar.post_content}
    coordinate={getBarCoordinate(bar)}
    image={chooseBeerIcon(bar.post_category)}
    onPress={onPress}
  >
    <MapView.Callout tooltip={true} />
  </MapView.Marker>
);

BarMarker.propTypes = {
  bar: PropTypes.shape({
    post_title: PropTypes.string.isRequired,
    post_content: PropTypes.string,
    post_category: PropTypes.string.isRequired,
    post_latitude: PropTypes.number.isRequired,
    post_longitude: PropTypes.number.isRequired,
  }).isRequired,
  onPress: PropTypes.func.isRequired,
};

export default BarMarker;

// Local Variables:
// mode: js2-jsx
// End:
