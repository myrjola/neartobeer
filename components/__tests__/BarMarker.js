// Import both of these before the test renderer
import React from 'react';
import 'react-native';

import renderer from 'react-test-renderer';

import BarMarker from '../BarMarker';

const exampleBar = {
  post_id: 38,
  post_title: 'MacLarens',
  post_author: 1,
  post_content: 'Serves chilled Norrlands Guld on tap for only 29:-.' +
    'One of the most best places in town for cheap beer lovers.',
  post_category: 'Beer under 30',
  default_category: 'Beer under 30',
  post_type: 'gd_place',
  post_status: 'publish',
  post_address: 'Norrtullsgatan 11',
  post_city: 'Stockholm',
  post_region: 'Stockholm County',
  post_country: 'Sweden',
  post_zip: '113 29',
  post_latitude: 59.3419634,
  post_longitude: 18.051719899999966,
};

const moreExpensiveBar = {
  post_id: 39,
  post_title: 'Lion Bar Sveav채gen 39',
  post_author: 1,
  post_content: '',
  post_category: 'Beer under 40',
  default_category: 'Beer under 30',
  post_type: 'gd_place',
  post_status: 'publish',
  post_address: 'Sveav채gen 39 Stockholm',
  post_city: 'Stockholm',
  post_region: 'Stockholm County',
  post_country: 'Sweden',
  post_latitude: 59.3404626,
  post_longitude: 18.05888660000005,
};

const mostExpensiveBar = {
  post_id: 39,
  post_title: 'Lion Bar Sveav채gen 39',
  post_author: 1,
  post_content: 'The famous Lion Bar chain is a perfect find for bargain brew lovers! 7 Places all around' +
    ' Stockholm serve chilled Spendrups (40cl, tapped) for 23:- until 20h. After this the prices will increase ' +
    'in two steps: until midnight the price will be 36:-, thereafter 40:-. Go and get there early to get yourself' +
    ' a crispy brew for under 30:- !',
  post_category: 'Beer under 50',
  default_category: 'Beer under 30',
  post_type: 'gd_place',
  post_status: 'publish',
  post_address: 'Sveav채gen 39 Stockholm',
  post_city: 'Stockholm',
  post_region: 'Stockholm County',
  post_country: 'Sweden',
  post_latitude: 59.3404626,
  post_longitude: 18.05888660000005,
};

it('renders', () => {
  expect(renderer.create(
    <BarMarker bar={exampleBar} onPress={jest.fn()} />,
  )).toMatchSnapshot();
});

it('renders with yellow icon', () => {
  expect(renderer.create(
    <BarMarker bar={moreExpensiveBar} onPress={jest.fn()} />,
  )).toMatchSnapshot();
});

it('renders with red icon', () => {
  expect(renderer.create(
    <BarMarker bar={mostExpensiveBar} onPress={jest.fn()} />,
  )).toMatchSnapshot();
});

// Local Variables:
// mode: react
// End:
