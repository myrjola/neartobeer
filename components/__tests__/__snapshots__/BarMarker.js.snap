// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Marker
  coordinate={
    Object {
      "latitude": 59.3419634,
      "longitude": 18.051719899999966,
    }
  }
  description="Serves chilled Norrlands Guld on tap for only 29:-.One of the most best places in town for cheap beer lovers."
  image="GreenMug.png"
  onPress={[Function]}
  title="MacLarens"
>
  <Callout
    tooltip={true}
  />
</Marker>
`;

exports[`renders with red icon 1`] = `
<Marker
  coordinate={
    Object {
      "latitude": 59.3404626,
      "longitude": 18.05888660000005,
    }
  }
  description="The famous Lion Bar chain is a perfect find for bargain brew lovers! 7 Places all around Stockholm serve chilled Spendrups (40cl, tapped) for 23:- until 20h. After this the prices will increase in two steps: until midnight the price will be 36:-, thereafter 40:-. Go and get there early to get yourself a crispy brew for under 30:- !"
  image="RedMug.png"
  onPress={[Function]}
  title="Lion Bar Sveavägen 39"
>
  <Callout
    tooltip={true}
  />
</Marker>
`;

exports[`renders with yellow icon 1`] = `
<Marker
  coordinate={
    Object {
      "latitude": 59.3404626,
      "longitude": 18.05888660000005,
    }
  }
  description=""
  image="YellowMug.png"
  onPress={[Function]}
  title="Lion Bar Sveavägen 39"
>
  <Callout
    tooltip={true}
  />
</Marker>
`;
